Code/classy_hrm.py

```diff
@@ -0,0 +1,19 @@
+def classy_hrm(file='ecg_data1.csv', window=20, brady_max=40, tachy_min=100):
+    """Main function to be used with class-based approach
+
+    :param file: str, name of input csv file
+    :param window: int, window size in sec to divide signal into for diagnosis
+    :param brady_max: int, upper threshold for bradycardia
+    :param tachy_min: int, lower threshold for tachycardia
+    :return: none
+    """
+    from ecg import ECG
+
+    data = ECG(file, window, brady_max, tachy_min)
+    data.write_file()
```

Comment:
this should perhaps indicate that there's some processing involved here, otherwise to a casual reader of this thread of code it looks like the data inside `data` is just itself being written to file. Perhaps there should be a `data.analyze` step or this function should be renamed to include analysis

Reply:
ah ha, I see that you're doing the analysis right on `init`...that's fine, but generally it's nice to have a transparent API to a class wrapper so a user generally knows what's going on. Typically large processing tasks won't happen in an init for a wrapper class, just basic things like type checking, unit normalization, etc. and then all following actions are generally explicit.